modules/core/client/components/Map/MapStyleControl.js

```diff
@@ -1,80 +1,126 @@
+            aria-expanded={isOpen}
+            aria-haspopup="true"
+            className="btn"
+            onClick={this.open}
+            onMouseEnter={() => !this._context.isDragging && this.open()}
+            aria-label={t('Change map style')}
+          >
             <MapIcon
               mapboxStyle={
-                selectedStyle !== MAP_STYLE_OSM.name ? selectedStyle : false
+                selectedStyle !== MAP_STYLE_OSM.name ? selectedStyle : ''
               }
-              mapboxToken={mapboxToken}
             />
             {mapStyle?.name || mapboxStyleNames[mapStyle]}
-          </>
-        }
-        id="map-style-picker"
-      >
-        {[
-          MAP_STYLE_MAPBOX_STREETS,
-          MAP_STYLE_MAPBOX_SATELLITE,
-          MAP_STYLE_MAPBOX_OUTDOORS,
-        ].map(mapboxStyle => (
-          <MenuItem
-            active={selectedStyle === mapboxStyle}
-            disabled={!mapboxToken}
-            key={mapboxStyle}
-            onClick={() => setMapstyle(mapboxStyle)}
-          >
-            <MapIcon mapboxStyle={mapboxStyle} mapboxToken={mapboxToken} />
-            {mapboxStyleNames[mapboxStyle]}
-          </MenuItem>
-        ))}
-        {process.env.NODE_ENV !== 'production' && (
-          <MenuItem
-            active={selectedStyle === MAP_STYLE_OSM.name}
-            key={MAP_STYLE_OSM.name}
-            onClick={() => setMapstyle(MAP_STYLE_OSM)}
+          </button>
+        )}
+        {isOpen && (
+          <div
+            className="btn-group-vertical"
+            onMouseLeave={this.close}
+            role="group"
           >
-            <MapIcon />
-            {MAP_STYLE_OSM.name}
-          </MenuItem>
+            {[
+              MAP_STYLE_MAPBOX_STREETS,
+              MAP_STYLE_MAPBOX_OUTDOORS,
+              MAP_STYLE_MAPBOX_SATELLITE,
+            ].map(mapboxStyle => (
+              <button
+                className={classnames('btn', 'btn-default', {
+                  'is-active': selectedStyle === mapboxStyle,
+                })}
+                disabled={!MAPBOX_TOKEN}
+                key={mapboxStyle}
+                onClick={() => {
+                  this.close();
+                  setMapstyle(mapboxStyle);
+                }}
+                type="button"
+              >
+                <MapIcon mapboxStyle={mapboxStyle} />
+                {mapboxStyleNames[mapboxStyle]}
+              </button>
+            ))}
+            {process.env.NODE_ENV !== 'production' && (
+              <button
+                className={classnames('btn', 'btn-default', {
+                  'is-active': selectedStyle === MAP_STYLE_OSM.name,
+                })}
+                key={MAP_STYLE_OSM.name}
+                onClick={() => {
+                  this.close();
+                  setMapstyle(MAP_STYLE_OSM);
+                }}
+                type="button"
+              >
+                <MapIcon />
+                {MAP_STYLE_OSM.name}
+              </button>
```

Comment:
Possible candidate for a separate component?

Reply:
Yup, I wasn't entirely sure about that since it's not really useful independently and it doesn't really affect line count — but I split it for now.

Thinking if to just remove OSM layer for now and that would remove the need for separate component here, too.

https://github.com/Trustroots/trustroots/pull/1329/commits/d3497344fe6ae55230765523dc4c51f9a72da33b